Rails 5 WIP

damianlegawiec opened the pull request.

Commit:
[Rails 5] view handler default changed from erb to raw

Commit:
[Rails 5] remove gem test_after_commit

Commit:
[Rails 5] add gem rails-controller-testing

Commit:
[Rails 5] explicitly throw(:abort) in callbacks to rollback

Commit:
[Rails 5] replace deprecated table_exists? with data_source_exists?

Commit:
[Rails 5] replace deprecated serve_static_files and static_cache_control with public_file_server.enabled and public_file_server.headers respectively

Commit:
[Rails 5] add silence_stream private method for install_generator, add silence_stream method for testing_support

Commit:
[Rails 5] replace Type::Integer.new with ActiveRecord::Type::Integer.new to move in ActiveRecord namespace

Commit:
[Rails 5] remove undefined callback call on StockItem

Commit:
[Rails 5] change #process method call args for upgraded rails

Commit:
[Rails 5] Arel::SelectManager accepts 1 arg now

Commit:
[Rails 5] fix rails 5 spec failures

Commit:
[Rails 5] replace ActiveRecord::Base with ApplicationRecord

Commit:
[Rails 5] [Spree Backend] replace redirect_to(:back) with redirect_back

Commit:
[Rails 5] remove only option from rescue_from as it was ignored before and not throws ArgumentError

Commit:
Need to explicitly use .id

Commit:
[Rails 5] remove terminator from define callbacks

Commit:
Monkey patch rabl for rails 5

Commit:
[Rails 5] fix postgress failing specs

Commit:
Give populate_redirect its own controller action

For some reason setting the flash in the inline redirect doesn't work in
rails 5.

Commit:
[Rails 5] remove maximum length validator spec

Commit:
Sanitize attrs earlier in Importer::Product

damianlegawiec committed:
Rails 5.0.0 dependencies met

damianlegawiec committed:
Specify Rails version for all migrations

ActiveRecord::Migration without a Rails version is deprecated since
Rails 5.

damianlegawiec committed:
Rails 5 requires Ruby >= 2.2.2

damianlegawiec committed:
Render :nothing is deprecated in Rails 5

damianlegawiec committed:
Silencing Rails 5 warnings in OrderContents#add_to_line_item

damianlegawiec committed:
Checkout spec permitted attributes fixes

damianlegawiec committed:
In Rails 5 ActionController::Parameters do not inherit from Hash

damianlegawiec committed:
Fixes ProductsController#redirect_if_legacy_path for Rails 5

We want to permit all parameters here in this redirect so any marketing
information, google analytics tags, referrals and so on are preserved.

damianlegawiec committed:
Fixes Spree::Core::Search::Base#add_search_scopes for Rails 5

In Rails 5 ActionController::Parameters Now Returns an Object Instead
of a Hash

damianlegawiec committed:
Fixes admin panel's per page dropdown for Rails 5

damianlegawiec committed:
Fixes deprecation warning for using merge! on params

`merge!` is deprecated and will be removed in Rails 5.1, as
`ActionController::Parameters` no longer inherits from hash

damianlegawiec committed:
Fixes API deprecation warning ActionController::TestCase HTTP request methods will accept only keyword arguments

damianlegawiec committed:
Fixes Return Authorisztions screens for Rails 5

AC::Params will not return hash by default in Rails 5

damianlegawiec committed:
Use keyword arguments in API specs

ActionController::TestCase HTTP request methods without keyword params
is deprecated.

damianlegawiec committed:
Silence deprecation warnings for Spree::Api::BaseController#map_nested_attributes_keys for Rails 5

AC::Parameters will not return hash by default in Rails 5

damianlegawiec committed:
Update RABL for Rails 5

damianlegawiec committed:
Fixes ActiveModel::Errors#[]= is deprecated warning message in Rails 5

damianlegawiec committed:
Fixes API specs for params returned as strings for Rails 5

In Rails 5 params values are aways converted to strings

damianlegawiec committed:
Fixes API::V1::ProductsController spec for Rails 5

In Rails 5 empty hash isn’t sent as params

damianlegawiec committed:
InventoryUnits API endpoint fixes for Rails 5

In Rails 5 belongs_to association adds presence validation by default.
Also sending an empty hash as params will not work.

damianlegawiec committed:
API spec fixes for new AC:Parameters behaviour in Rails 5

novotarq committed:
Model fixes for new AC:Parameters behaviour in Rails 5

tanmay3011 commented on backend/app/views/spree/admin/orders/cart.html.erb:
```diff
@@ -2,6 +2,8 @@
 
 <%= render 'spree/admin/shared/order_tabs', current: :cart %>
 
+<h1><%= 'ABC' %></h1>
```
@damianlegawiec Does this 'ABC' relates to something? I guess this is left by mistake or some other text needs to be here.

damianlegawiec replied:
@tanmay3011 fixed! Thanks for noticing :)

damianlegawiec committed:
Admin::UsersController spec fixes for Rails 5

AC::Parameters in Rails 5 won’t return a hash

damianlegawiec committed:
Admin::Order::CustomerDetailsController spec fixes for Rails 5

damianlegawiec committed:
Fixes some funky per_page dropdown specs

damianlegawiec committed:
Grab deleted variants from line item

Same as for inventory unit
(https://github.com/spree/spree/commit/4c1cdb0880386323d7878f3bd41af21e0
cb641a5)

Required for Rails 5 belongs_to presence validation

damianlegawiec committed:
Fixes Orders sample for Rails 5

first_or_create leaks scope and can lead to strange behaviour
